tests/torch/quantization/test_overflow_issue_export.py

```diff
@@ -433,3 +434,77 @@ def test_is_overflow_fix_applied_model_resumed_correctly(tmp_path):
         model, nncf_config, compression_state=compression_state)
     load_state(compressed_model, model_state_dict, is_resume=True)
     are_symmetric_fq_nodes_are_exported_correct_with_overflow_fix(tmp_path, compression_ctrl)
+
+
+@pytest.mark.parametrize("quantization_mode", ["symmetric", "asymmetric"])
+def test_overflow_fix_quantization_export_with_middle_quants(quantization_mode):
+    model = nn.Sequential(nn.Linear(in_features=128, out_features=100, bias=False))
+    sample_size = [1, 1, 100, 128]
+    config = NNCFConfig()
+    config.update({
+        "input_info": {
+            "sample_size": sample_size
+        },
+        "compression": {
+            "algorithm": "quantization",
+            "weights": {
+                "mode": quantization_mode
+            },
+            "export_to_onnx_standard_ops": False
+        }
+    })
+    register_bn_adaptation_init_args(config)
+
+    def set_scale_to_sym_quantizer_and_get_attrs(quantizer: SymmetricQuantizer, scale: float):
+        scale = np.full(quantizer.scale.size(), scale)
+        levels = quantizer.levels
+        level_low = quantizer.level_low
+        level_high = quantizer.level_high
+        input_low = scale * (level_low / level_high)
+        input_range = scale - input_low
+        quant_len = input_range / (levels - 1)
+        quantizer.scale = nn.Parameter(torch.from_numpy(scale.astype(np.single)))
+        return input_low, quant_len, levels
+
+    def set_scale_to_asym_quantizer_and_get_attrs(quantizer: AsymmetricQuantizer, input_low: float, input_range: float):
+        input_low = np.full(quantizer.input_low.size(), input_low)
+        input_range = np.full(quantizer.input_low.size(), input_range)
+        levels = quantizer.levels
+        quant_len = input_range / (levels - 1)
+        quantizer.input_low = nn.Parameter(torch.from_numpy(input_low.astype(np.single)))
+        quantizer.input_range = nn.Parameter(torch.from_numpy(input_range.astype(np.single)))
+        return input_low, quant_len, levels
+
+    weights_size = list(model[0].weight.size())
+    compressed_model, compression_ctrl = create_compressed_model_and_algo_for_test(model, config)
+    compressed_model.eval()
+    weight_quantizer = list(compression_ctrl.weight_quantizers.values())[0].quantizer_module_ref
+
+    if quantization_mode == "symmetric":
+        input_low, quant_len, levels = set_scale_to_sym_quantizer_and_get_attrs(weight_quantizer, scale=1.)
```

Comment:
please lightweight the test body
- move nested functions outside
- remove if for symmetric by parametrizing test case by type and set_scale_fn
- reuse get_config_for_export_mode
- extract ref weights generation to an outside function
- think about other improvements to make it more readable and simple

Reply:
Done, take a look at the changed test